runtime/scripts/parts/patternmatch.js

```diff
@@ -114,24 +117,53 @@ PatternMatchPart.prototype = /** @lends Numbas.PatternMatchPart.prototype */ {
 		}
 		this.answered = this.studentAnswer.length>0;
 
-		var caseInsensitiveAnswer = new RegExp( this.settings.correctAnswer, 'i' );			
-		var caseSensitiveAnswer = new RegExp( this.settings.correctAnswer );
-		
-		if( this.settings.caseSensitive ) {
-			if( caseSensitiveAnswer.test(this.studentAnswer) ) {
-				this.setCredit(1,R('part.marking.correct'));
-			} else if(caseInsensitiveAnswer.test(this.studentAnswer)) {
-				this.setCredit(this.settings.partialCredit,R('part.patternmatch.correct except case'));
+		var caseInsensitiveRegexAnswer = new RegExp( this.settings.correctAnswer, 'i' );			
```

Comment:
There's no point in collecting both the regex and non-regex correct answers. Do a switch statement to decide which marking algorithm to use:

    switch(this.settings.matchMode) {
        case 'regex':
            // the old code
            break;
        case 'exact':
            // compare this.studentAnswer to this.settings.correctAnswer with `==`, instead of a regex test
            // if they don't match, convert both to lowercase and compare again, then apply the partial credit if necessary
    }